geonode/resource/manager.py

```diff
@@ -764,13 +728,15 @@ def assignable_perm_condition(perm, resource_type):
                             _prev_perm = _perm_spec["users"].get(_owner, []) if "users" in _perm_spec else []
                             _perm_spec["users"][_owner] = set.union(perms_as_set(_prev_perm), perms_as_set(['change_dataset_data', 'change_dataset_style']))
 
-                        _resource.handle_moderated_uploads()
+                        _resource = AdvancedSecurityWorkflowManager.handle_moderated_uploads(_resource.uuid, instance=_resource)
```

Comment:
should this be moved at the end of the code-block and let it handle both cases?

Reply:
done